### DISCUSSION TITLE
Can godot values be initialised inside of Go init() functions?

### DISCUSSION THREAD

**Discussion by ardawan** [quaadgras/graphics.gd#9] [Q&A] [answered]
wondering if we can add some placeholders or variables to the editor's placeholder for each scene via code specifically in init() function?
Then we run `gd` those settings load to the editor.

e.g `var env1 = preload("res://env2.tres")`

**Comment by Splizard** [comment, marked as answer]
Most `gd` package functionality is invalid before `gdextension.Link` is called, as this is what connects everything in the package up to Godot. You're probably calling this in `main` which is after `init` so `gd` functions won't work inside a `init` function. Even if you made sure to call `gdextension.Link` beforehand, not everything from Godot is initialised when Go starts up. 

If possible, perform initialisation inside a Node's `Ready` implementation. There isn't a preload equivalant yet, I'll have a look at this, it will probably be based on struct tagging if added. At this stage. A `gd.ResourceLoader` can be added to your struct, it's a singleton so it will be instantiated for you and I believe it has a `Load` method which is what `preload` uses.

**Reply by ardawan** [reply]
Thanks for the clarification.

I was asking that because I thought we may have something equivalent for the `onready` in Gdscript.

Can we have some examples of how can we load resources via this package?

**Reply by Splizard** [reply]
@ardawan 
I've just added a `Load` function to the `Resource` classdb package, you should be able to use it like this:
```go
    var env1 = Resource.Load[WorldEnvironment.Instance]("res://env2.tres")
```
